### PR TITLE
Switch from the NoFork terminology to the TerraformPluginSDK terminology and bump dependiencies

### PR DESCRIPTION
**Pull request by sergenyalcin**
### Description of your changes

In this PR:
- Switch from the NoFork terminology to the TerraformPluginSDK terminology
- Bump crossplane-runtime version to `v1.15.1`
- Bump upjet version to `v1.1.1`

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- `examples/redis/instance.yaml` => https://github.com/upbound/provider-gcp/actions/runs/7990615683
- `examples/compute/network.yaml` => https://github.com/upbound/provider-gcp/actions/runs/7990599624
- `examples/compute/route.yaml` => https://github.com/upbound/provider-gcp/actions/runs/7990605824
- `examples/cloudplatform/serviceaccount.yaml` => https://github.com/upbound/provider-gcp/actions/runs/7990590775
- `examples/storage/bucketobject.yaml` => https://github.com/upbound/provider-gcp/actions/runs/7990630999
- `examples/bigquery/connection.yaml` =>
```shell
NAME                                            READY   SYNCED   EXTERNAL-NAME
                                     AGE
connection.bigquery.gcp.upbound.io/connection   True    True     projects/official-provider-testing/locations/US/connections/31
8e9757-4747-4f89-98e4-a144e20cfec7   47m

NAME                             READY   SYNCED   EXTERNAL-NAME   AGE
database.sql.gcp.upbound.io/db   True    True     db              47m

NAME                                                          READY   SYNCED   EXTERNAL-NAME             AGE
databaseinstance.sql.gcp.upbound.io/example-instance-test12   True    True     example-instance-test12   58m

NAME                           READY   SYNCED   EXTERNAL-NAME   AGE
user.sql.gcp.upbound.io/user   True    True     user            47m
```

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/redis/instance.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/compute/network.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/bigquery/connection.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/storage/bucketobject.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/bigquery/connection.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/bigquery/connection.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/bigquery/connection.yaml"

File: go.mod [hunk 1]
```diff
@@ -115,54 +117,64 @@ require (
...
+
+replace golang.org/x/exp => golang.org/x/exp v0.0.0-20230713183714-613f0c0eb8a1
+
+replace github.com/hashicorp/terraform-json => github.com/hashicorp/terraform-json v0.17.0
+
+replace github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/hashicorp/terraform-plugin-sdk/v2 v2.26.1
+
+replace github.com/hashicorp/terraform-plugin-go => github.com/hashicorp/terraform-plugin-go v0.16.0
+
+replace github.com/hashicorp/terraform-plugin-framework => github.com/hashicorp/terraform-plugin-framework v1.2.0
```
Comment (turkenf) [inline go.mod:180]: Let's fix the versions to the ones already defined in `go.mod`:

```suggestion
replace golang.org/x/exp => golang.org/x/exp v0.0.0-20231110203233-9a3e6036ecaa

replace github.com/hashicorp/terraform-json => github.com/hashicorp/terraform-json v0.14.0

replace github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/hashicorp/terraform-plugin-sdk/v2 v2.24.0

replace github.com/hashicorp/terraform-plugin-go => github.com/hashicorp/terraform-plugin-go v0.14.3

```

**Review by turkenf — APPROVED** [review]
Thank you @sergenyalcin, LGTM.

File: go.mod (hunk 1 above)
Comment (sergenyalcin) [reply go.mod]: Done!

**Pull request merged**